doc/modules/model_evaluation.rst

```diff
@@ -258,6 +258,122 @@ permitted and will require a wrapper to return a single metric::
+median                absolute error              strictly consistent
+mode                  zero-one loss               consistent (for binary classification)
+**Regression**
+mean                  squared error               strictly consistent (if finite 2nd moment)
+mean                  Poisson deviance            strictly consistent (for non-negative target y)
+mean                  Gamma deviance              strictly consistent (for positive target y)
+median                absolute error              strictly consistent
+mode                  zero-one loss               asymptotically consistent
+==================    ========================    ===============================================
+
+The zero-one loss is equivalent to the accuracy score, meaning it gives
+different score values but the same ranking.
+R² is equivalent to squared loss.
+The Brier score is the squared error for classification.
+
+An example with binary classification
+-------------------------------------
+
+Suppose you want to predict whether an apple blossom in spring becomes an
+eatable apple fruit in autumn.
+This is binary classification for :math:`Y = \text{fruit} = 1` or
+:math:`Y = \text{no fruit} = 0`.
+The good thing about binary classification is that predicting the mean of
+:math:`Y`, :math:`\operatorname{E}[Y] = p` (a result of setting the values of
+:math:`Y` to 0 and 1), is equivalent to predicting the whole distribution
+:math:`F(Y=\text{fruit}) = \operatorname{Pr}(Y=\text{no fruit}) = p`.
+Therefore, a good scoring function is one that is strictly consistent for the
+mean, e.g. log loss or Brier score.
+Note that for binary classification, the mode - the most probable outcome
+(*fruit* if :math:`p>0.5`) -
+is less informative than the mean: To say that we expect a fruit is less
+informative than to say that we expect a fruit with probability :math:`p`.
+
+On the other side, depending on the purpose of your prediction, you might
+also be interested in the use of other scoring functions:
+
+* If you want to optimize the conditions for good blossoms and you therefore
+  cut of blossoms for which you predict *no fruit*, you might want to optimize
+  the false negative rate (predicting *no fruit* when it actually gives a
+  fruit) in order to avoid cutting of too many.
+* If you sell your apples in advance base on the predicted number of fruits,
+  you might want to avoid selling too much and therefore go for the false
+  positive rate (predicting *fruit* when it actually does not give a fruit).
+* Treating the target variable :math:`y` as a random variable may just be an
+  approximation, but the true nature might be deterministic.
+  For example, if there was no bee to pollinate the blossoms, there won't be
+  any fruit--with certainty.
+  Some scoring functions like the log loss have problems at the boundaries
+  :math:`\hat{y}=0` or :math:`\hat{y}=1` (the same might be
+  true for an estimator).
```

Comment:
This suggestions use :math:`\text{fruit}` and :math:`\text{no fruit}` for consistency.

```suggestion
Therefore, a good scoring function is one that is strictly consistent for the
mean, e.g. log loss or Brier score.
Note that for binary classification, the mode - the most probable outcome
(:math:`\text{fruit}` if :math:`p>0.5`) -
is less informative than the mean: To say that we expect a fruit is less
informative than to say that we expect a fruit with probability :math:`p`.

On the other side, depending on the purpose of your prediction, you might
also be interested in the use of other scoring functions:

* If you want to optimize the conditions for good blossoms and you therefore
  cut of blossoms for which you predict :math:`\text{no fruit}`, you might want 
  to optimize the false negative rate (predicting :math:`\text{no fruit}` when
  it actually gives a fruit) in order to avoid cutting of too many.
* If you sell your apples in advance base on the predicted number of fruits,
  you might want to avoid selling too much and therefore go for the false
  positive rate (predicting :math:`\text{fruit}` when it actually does not give a
  fruit).
* Treating the target variable :math:`y` as a random variable may just be an
  approximation, but the true nature might be deterministic.
  For example, if there was no bee to pollinate the blossoms, there won't be
  any fruit--with certainty.
  Some scoring functions like the log loss have problems at the boundaries
  :math:`\hat{y}=0` or :math:`\hat{y}=1` (the same might be
  true for an estimator).
```